File: Python/2016/Arrival Time.py
# J4

# Rush hour --> 7(420 mins)-10(600 mins) and  3(900 mins)-7(1140 mins)

hour, minute = list(map(int, input().strip().split(':')))
left = hour * 60 + minute
arrived = left

for minute in range(left, left + 120):
    if arrived >= 420 and arrived < 600 or arrived >= 900 and arrived < 1140:
        arrived += 2
    else:
        arrived += 1

    if arrived == 1400: # 1400 mins make up 24 hrs
        arrived = 0


hours, minutes = 0, 0

while (arrived >= 60):
    arrived -= 60
    hours += 1
    if arrived < 60:
        break

minutes = arrived # whatever is left in arrived (will be < 60)

# formatting

if hours < 10:
    hours = '0' + str(hours)

if minutes < 10:
    minutes = '0' + str(minutes)

print(str(hours) + ":" + str(minutes))

File: Java/2016/ArrivalTime.java
// J4

import java.util.Arrays;
import java.util.Scanner;

public class ArrivalTime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.next();

        int[] time = Arrays.stream(input.split(":")).mapToInt(Integer::parseInt).toArray();

        int left = time[0] * 60 + time[1];

        int arrived = left;

        for (int minute = left; minute < left + 120; minute++) {
            if (arrived >= 420 && arrived < 600 || arrived >= 900 && arrived < 1140) arrived += 2;
            else arrived++;

            if (arrived == 1400) arrived = 0;
        }

        int hours = 0, minutes = 0;

        while (arrived >= 60) {
            arrived -= 60;
            hours++;
            if (arrived < 60) break;
        }

        minutes = arrived;

        String hours1, minutes1;

        if (hours < 10) hours1 = "0" + Integer.toString(hours); else hours1 = Integer.toString(hours);
        if (minutes < 10) minutes1 = "0" + Integer.toString(minutes); else minutes1 = Integer.toString(minutes);

        System.out.println(hours1 + ":" + minutes1);
        
        scanner.close();
    }
}